test/metric_value_benchmark/metrics_value_benchmark_test.go

```diff
@@ -120,17 +119,33 @@ func shouldRunEC2Test(env *environment.MetaData, t *test_runner.TestRunner) bool
 	return ok
 }
 
-func isAllValuesGreaterThanOrEqualToZero(metricName string, values []float64) bool {
+// isAllValuesGreaterThanOrEqualToExpectedValue will compare if the given array is larger than 0
+// and check if the average value for the array is not la
+// TODO: Moving metric_value_benchmark to validator
+// https://github.com/aws/amazon-cloudwatch-agent-test/pull/162
+func isAllValuesGreaterThanOrEqualToExpectedValue(metricName string, values []float64, expectedValue float64) bool {
 	if len(values) == 0 {
-		log.Printf("No values found for: %v", metricName)
+		log.Printf("No values found %v", metricName)
 		return false
 	}
+
+	totalSum := 0.0
 	for _, value := range values {
 		if value < 0 {
-			log.Printf("Values are not all greater than or equal to zero for %v", metricName)
+			log.Printf("Values are not all greater than or equal to zero for %s", metricName)
 			return false
 		}
+		totalSum += value
+	}
+	metricErrorBound := 0.1
+	metricAverageValue := totalSum / float64(len(values))
+	upperBoundValue := expectedValue * (1 + metricErrorBound)
+	lowerBoundValue := expectedValue * (1 - metricErrorBound)
+	if expectedValue > 0 && (metricAverageValue > upperBoundValue || metricAverageValue < lowerBoundValue) {
```

Comment:
Why are we allowing for this variance?

Reply:
Glad you asked !!! CWA has two kinds of inputs:
* Normal inputs for host resources metrics (e.g `cpu`, `mem`)
* Service inputs for (e.g `stasd`, `collectd`)

For the first case, if we ever want to validate the host resources which we would tied up to the instances and will be flaky to do so. Therefore, we cannot validate exactly the value that tied to the instance size

For the second case, in case such as `statsd` has incremented value when being a counter or a sampling (sampling of counter), if the collection interval is 5 second, the counter will have the value of 5. However, this won't be true all the time since `counter` can be delayed when dealing with cache (e.g the counter value in the first 5 second will be 4 and the counter for the next 5 second will fill up the lost amount which is 6). So it will flaky too. Therefore, it would be flaky if consider high resolution metrics

Reply:
That is an implementation detail for the individual test runners. Changing the entire implementation across all of the test runners is unnecessary and changes our validation behavior.

Reply:
Hmm ! I would prefer generalization  that applied to all plugins but you are right on the fact that 
```
That is an implementation detail for the individual test runners.
```